### PR TITLE
Password change feature added

### LINKED ISSUE #522: Implement Change Password functionality
## Description
As a volunteer, I must have the option of changing my password.

## Mocks
image

## Acceptance Criteria
- [ ] create urls for inbuilt views of django.contrib.auth
- [ ] Template create for each step

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
30 minutes

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
# Description
The user can now change his password 

Fixes #522 

# Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
image
image

# Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak I have squashed the commits. Please review this too.

File: requirements.txt [hunk 1]
```diff
@@ -3,11 +3,12 @@ coveralls==0.5
 Django==1.7.4
 docopt==0.6.2
 flake8==3.4.1
-psycopg2==2.5.4
+psycopg2==2.7.3
 PyYAML==3.11
 requests==2.7.0
-selenium==2.53.1
+selenium==3.8.0
```
Comment (tapasweni-pathak) [inline requirements.txt]: :wave: Did Travis pass for your fork after this change?

Comment (anjali-dhanuka) [reply requirements.txt]: No it states that geckodriver is needed with this version of selenium.

File: vms/vms/settings.py [hunk 2]
```diff
@@ -22,14 +22,14 @@
...
 # Application definition
 INSTALLED_APPS = (
+    'authentication',
     'django.contrib.admin',
     'django.contrib.auth',
     'django.contrib.contenttypes',
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'administrator',
```
Comment (tapasweni-pathak) [inline vms/vms/settings.py]: 👌

File: vms/vms/settings.py [hunk 3]
```diff
@@ -22,14 +22,14 @@
 
 # Application definition
 INSTALLED_APPS = (
+    'authentication',
```
Comment (tapasweni-pathak) [inline vms/vms/settings.py]: 👌

File: requirements.txt (hunk 1 above)
Comment (tapasweni-pathak) [reply requirements.txt]: thought so, then we revert this change.

Comment (anjali-dhanuka) [reply requirements.txt]: Should I change selenium version back to 2.53.1?

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak Please have a look at it. I have changed the selenium version back to the old one.

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak Please have a look at it

File: vms/authentication/templates/registration/password_change_form.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{% extends "vms/base.html" %}
+{% block content %}
+<div id="content-main">
+
+<form action="" method="post">{% csrf_token %}
+<div>
+{% if form.errors %}
+    <p class="errornote">
+    {% if form.errors.items|length == 1 %}Please correct the error below.{% else %}Please correct the errors below.{% endif %}
```
Comment (tapasweni-pathak) [inline vms/authentication/templates/registration/password_change_form.html]: You may would want to indent these.

File: vms/authentication/templates/registration/password_change_form.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+</div>
+
+<div class="form-row">
+    {{ form.new_password1.errors }}
+    <label for="id_new_password1" class="required">New password: </label>{{ form.new_password1 }}
+</div>
+
+<div class="form-row">
+{{ form.new_password2.errors }}
+    <label for="id_new_password2" class="required">Password (again): </label>{{ form.new_password2 }}
```
Comment (tapasweni-pathak) [inline vms/authentication/templates/registration/password_change_form.html]: can you please share what a `password2` is here?

File: vms/authentication/templates/registration/password_change_form.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+</div>
+
+</fieldset>
+<div class="submit-row">
+    <input type="submit" value="Change my password" class="default" />
+</div>
+
+<script type="text/javascript">document.getElementById("id_old_password").focus();</script>
```
Comment (tapasweni-pathak) [inline vms/authentication/templates/registration/password_change_form.html]: o cool.

File: vms/authentication/templates/registration/password_change_form.html (hunk 4 above)
Comment (anjali-dhanuka) [reply vms/authentication/templates/registration/password_change_form.html]: Done with indenting and rebasing. Please review.

**Comment by anjali-dhanuka** [comment]
@tonythomas01 @tapasweni-pathak 
I have rebased and resolved all the conflicts as well. Please review

File: vms/authentication/templates/registration/password_change_done.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+<div class="breadcrumbs">
+<a href="{% url 'admin:index' %}">Home</a>
+Password change
+</div>
+{% endblock %}
+
+{% block title %}{{ title }}{% endblock %}
+{% block content_title %}<h1>{{ title }}</h1>{% endblock %}
+{% block content %}
+<p>Your password was changed.</p>
```
Comment (tapasweni-pathak) [inline vms/authentication/templates/registration/password_change_done.html]: VMS would be merged w/ Portal going forward, could you look for messages in Portal for changing password and update these. Also you need to load i18n file for enabling translation.

File: vms/authentication/templates/registration/password_change_form.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
+{% extends "vms/base.html" %}
+{% block content %}
+<div id="content-main">
+
+<form action="" method="post">{% csrf_token %}
+<div>
+{% if form.errors %}
+<p class="errornote">
+{% if form.errors.items|length == 1 %}Please correct the error below.{% else %}Please correct the errors below.{% endif %}
```
Comment (tapasweni-pathak) [inline vms/authentication/templates/registration/password_change_form.html]: we can remove the else here.

File: vms/authentication/templates/registration/password_change_done.html (hunk 7 above)
Comment (anjali-dhanuka) [reply vms/authentication/templates/registration/password_change_done.html]: @tapasweni-pathak The portal does not show up any message on password change. It just reloads the password change form. I think that the portal should come up with this feature.

**Comment by ayushgarg1804** [comment]
As @kamsuri said, please follow the PR format.

**Comment by anjali-dhanuka** [comment]
@kamsuri @ayushgarg1804 PR template has been implemented. Please review.

**Comment by kamsuri** [comment]
@anjali-dhanuka please squah the commits.

**Comment by anjali-dhanuka** [comment]
@kamsuri Done with the squashing. Please review.

**Comment by anjali-dhanuka** [comment]
Closing, PR exists in gsoc18-code https://github.com/systers/vms/pull/801

**Pull request closed without merging**